feat: prompt users to add GitHub token when report returns no data

singhmahipal opened the pull request:
Fixes #277

### What
Adds a user-facing prompt when report generation returns no data due to a missing GitHub Personal Access Token.

### Why
New users currently receive no feedback when generating reports without a token, which leads to confusion during onboarding.

### Screenshots / Video
- Before: 
<img width="1919" height="1079" alt="Screenshot 2025-12-23 140059" src="https://github.com/user-attachments/assets/f72de885-d593-464b-85c1-aab3bcd69106" />

- After: 

https://github.com/user-attachments/assets/5790cb35-a0e0-4dd7-ac88-45d202293798



### Notes
`npm run format` / `npm run check` currently fail due to an upstream Biome schema/CLI mismatch. No formatting or tooling changes are included in this PR.

## Summary by Sourcery

Add onboarding guidance and UI improvements to the scrum report popup when GitHub data cannot be fetched.

New Features:
- Show a localized onboarding message explaining why a GitHub token is required and guiding users to settings and GitHub token creation when no token is present.
- Allow users to enter and persist an optional "Your Reason" text for the generated report from the popup UI.

Enhancements:
- Improve error and invalid-token messaging in the popup using DOM-based elements instead of raw HTML strings, and consistently restore the Generate Report button state.
- Reuse stored user-specific reason text when generating reports instead of clearing it from storage.
- Harden popup initialization by safely reading platform-specific usernames from stored settings.

singhmahipal commented:
All requested security fixes and refactor feedback have been addressed in the latest commit.
The PR is now ready for final review and merge.

Thanks again for the thorough review and security feedback 🙌

singhmahipal commented:
Thanks for the clarification, @vedansh-5 .
I understand the vision now — the extension should allow users to generate and inject reports with basic functionality without requiring an auth token initially.

I’ll update the flow so the token UI is prompted only in the following cases:

When the user exceeds the free API limit

When the user explicitly wants to access token-based features/filters

I’ll also improve the token UI as suggested.

I’ll push the updated changes shortly for re-review.

singhmahipal commented:
Hi @vedansh-5  , 

The requested changes have been implemented:

- Basic report generation works **without requiring an auth token**.
- Token UI now appears **only** in these cases:
  1. Free API limit is exceeded
  2. User tries to access token-based features/filters
- Token UI messaging and UX improved for clarity.

Here's a short video demonstrating the updated flow:  

https://github.com/user-attachments/assets/29987470-2ceb-49a5-8bb8-f9315d6990dc

Kindly review the updated changes.

singhmahipal commented:
Hi! 👋

I’ve updated the UI to better match the existing design:

Aligned colors with the current UI palette

Reduced excessive spacing and padding

Simplified borders to match the rest of the extension

Added a screenshot preview below for reference.
<img width="466" height="748" alt="Screenshot 2025-12-31 144707" src="https://github.com/user-attachments/assets/dbd39ddc-2e36-462b-8c2d-b5348ae5b438" />


Please let me know if you’d like any further tweaks — happy to adjust.

Thanks!

vedansh-5 reviewed (changes requested):
<img width="461" height="681" alt="image" src="https://github.com/user-attachments/assets/99e06af1-b08f-4304-acd9-bccfc3fce693" />
Please adjust for darkmode.
<img width="537" height="732" alt="image" src="https://github.com/user-attachments/assets/f1c6ed28-9593-466f-b25b-3b62dec3f669" />
Reduce this spacing, also could you add screenshots with a reduced border size, since there's already a border of extension window another thick border feels unnecessary, instead add a thin border. Thanks @singhmahipal

singhmahipal commented:
Hi @vedansh-5 ,
I’ve fixed the spacing issue — the report content now sits closer to the popup edges. 

📸 Screenshot:
<img width="466" height="754" alt="Screenshot 2026-01-03 153732" src="https://github.com/user-attachments/assets/73edd532-f017-4168-a582-1d0705008a5f" />

Additionally, I’ve adjusted the dark mode issue due to which the text was not properly visible. 

Let me know if you want any changes again.
Thanks!!!

singhmahipal commented:
Hi, 
✅ Fixes applied as requested:

- Thick border reduced to match UI style
- Background colors updated to match dark mode
- Rose color removed from background in light mode
- "Your Reason" section removed

Preview: 

- Rose color removed in light mode:
<img width="451" height="749" alt="Screenshot 2026-01-07 195037" src="https://github.com/user-attachments/assets/a04523bb-3a55-4d31-bf58-ab38bbdc03a3" />


- Dark mode background matched to UI:
<img width="453" height="741" alt="Screenshot 2026-01-07 195807" src="https://github.com/user-attachments/assets/b7f65603-93e7-493a-ba5a-690f497102bb" />

suhaib-ilahi commented:
> Closing this PR due to inactivity.
> 
> @suhaib-ilahi You can take this up. Please make sure to understand the scope clearly before working on it, the prompt should appear **when the user hits the hourly rate limit of the free GitHub API** .
> 

Sure. I would love to work on this.
